src/state/apis/swapper/swapperApi.ts

```diff
@@ -109,10 +108,28 @@ export const swapperApi = swapperApiBase.injectEndpoints({
               }
 
               const [isTradingActiveOnSellPool, isTradingActiveOnBuyPool] = await Promise.all(
-                [sellAsset.assetId, buyAsset.assetId].map(assetId => {
-                  return queryClient.fetchQuery(
-                    reactQueries.common.isTradingActive({ assetId, swapperName }),
-                  )
+                [sellAsset.assetId, buyAsset.assetId].map(async assetId => {
+                  const inboundAddresses = await queryClient.fetchQuery({
+                    ...reactQueries.thornode.inboundAddresses(),
+                    // Go stale instantly
+                    staleTime: 0,
+                    // Never store queries in cache since we always want fresh data
+                    gcTime: 0,
+                  })
+
+                  const inboundAddressResponse = selectInboundAddressData(inboundAddresses, assetId)
+
+                  const mimir = await queryClient.fetchQuery({
+                    ...reactQueries.thornode.mimir(),
+                    staleTime: thorchainBlockTimeMs,
+                  })
+
+                  return selectIsTradingActive({
+                    assetId,
+                    inboundAddressResponse,
+                    swapperName: SwapperName.Thorchain,
```

Comment:
Keeping it as-is for now as that was the previous flow and doing it a different way would risk breaking things, but I agree, it feels off.
Ideally, we should simply *not* run the query (i.e `enabled: swapperName === SwapperName.Thorchain`, which surely would give us `undefined` `data`, but since we have a hook abstracting this over, we could manage that there and simply return `true` for other swappers.

Added to the TODOs list, to be tackled in plane!